Feature/emoji categories 182132580

pererinha opened the pull request:
# Description
[reactions: emoji categories](https://www.pivotaltracker.com/n/projects/2561773/stories/182132580)

- implemented the new layout
- using new icon for categories
- vertical scroll
- implemented the emoji history
- if the user selects a emoji that he had already selected it will remove the reaction

# Impact

Describe impact, risks and mitigation if applicable.
  - Medium: UI changes that can block some actions but not the core features
  - 
# How to test
- go to the message menu page
- click at the plus button
- navigate between the emoji categories

![emoji-category](https://user-images.githubusercontent.com/99601/168202255-4abab27c-2225-4a85-9278-ffd94ab3ddd2.gif)

tgfassina commented on src/types/Emoji.ts:
```diff
@@ -1,28 +1,16 @@
-  short_names: string[];
   category: string;
-  subcategory: string;
-  sort_order: number;
```
I'm not sure what's this one for, but the order in which we're displaying is different from what I usually see. Maybe this helps?

pererinha replied:
kk, i just added them back, thanks @tgfassina

tgfassina commented:
@pererinha did you like the horizontal scroll? felt a bit weird to me, specially the fact that everything shows in columns instead of rows. I think everything I've used so far scrolls vertically, and in some apps the horizontal scroll switches between categories

pererinha commented:
> @pererinha did you like the horizontal scroll? felt a bit weird to me, specially the fact that every other app shows emoji in columns instead of rows. I think everything I've used so far scrolls vertically, and in some apps the horizontal scroll switches between categories

hey @tgfassina 
I changed the way it creates the grid with the emojis

pererinha commented:
hey @tgfassina 
i just finished some changes that we have talked about

- i did some refactor at the code
- made some improvements at the rendering (it should be faster now)
- fixed the scroll, you should be able to scroll between categories
- other tweaks

pls let me know if there is anything else i should do on this
thank you <3 

![emoji-category](https://user-images.githubusercontent.com/99601/168791085-b26fd2de-5fdd-4687-8db3-ed87eb1018d6.gif)

tgfassina commented:
Found a few problems. You can see this on the gif too: after you select a few reactions, they go into the fixed bar and push the +/× out of sight.

The selector is also not showing correctly on Android

pererinha commented:
hey @tgfassina 

i just fixed the pending stuff
- it is working on android
- i fixed button +/x
- i did the change to pre-load the next/prev page

pererinha commented on src/components/Emojis/EmojiCategoryView.tsx:
```diff
@@ -0,0 +1,128 @@
+// TODO:
+// handle the warning: Calling `getNode()` on the ref of an Animated component is no longer necessary.
+// You can now directly use the ref instead. This method will be removed in a future release., ReactNativeFiberHostComponent
+// this message is comming from the ScrollableTabView
```
i've created this chore for that https://www.pivotaltracker.com/n/projects/2561773/stories/182215121
so you can check that later
